include/jemalloc/internal/edata.h

```diff
@@ -185,22 +185,28 @@ struct edata_s {
 		size_t			e_bsize;
 	};
 
-	/*
-	 * List linkage, used by a variety of lists:
-	 * - bin_t's slabs_full
-	 * - extents_t's LRU
-	 * - stashed dirty extents
-	 * - arena's large allocations
-	 */
-	ql_elm(edata_t) ql_link;
-
-	/*
-	 * Linkage for per size class sn/address-ordered heaps, and
-	 * for extent_avail
-	 */
-	phn(edata_t)		ph_link;
+	union {
+		/*
+		 * List linkage used when the edata_t is active; either in
+		 * arena's large allocations or bin_t's slabs_full.
+		 */
+		ql_elm(edata_t)	ql_link_active;
+		/*
+		 * Pairing heap linkage.  Used whenever the extent is inactive
+		 * (in the page allocators), or when it is active and in
+		 * slabs_nonfull, or when the edata_t is unassociated with an
+		 * extent and sitting in an edata_cache.
+		 */
+		phn(edata_t)	ph_link;
+	};
```

Comment:
I ended up thinking no, just because we couldn't "really" catch any bugs (i.e. since they share storage, they'll share linkage, and so everything ends up working out the same).

You can imagine one day having debug info in lists that lets us track list<->node associations and assert on them; at that point it might be helpful, but we should probably also do that a little more systematically.